### ISSUE TITLE
ImDrawList::_OnChangedClipRect() called on an empty CmdBuffer when docking

### ISSUE THREAD

**Issue by yngccc** [ocornut/imgui#3311]
```
void ImDrawList::_OnChangedClipRect()
{
    // If current command is used with different settings we need to add a new command
    ImDrawCmd* curr_cmd = &CmdBuffer.Data[CmdBuffer.Size - 1];
    ...
}
```
`CmdBuffer.Size` is 0, this happens when I have window B docked inside window A then attempt to drag a window outside.
![ezgif com-crop](https://user-images.githubusercontent.com/1298221/85206329-4ef79e00-b2ef-11ea-9cef-538e654f7069.gif)

**Comment by ocornut** [comment]
Thank you, will investigate soon.

**Comment by ocornut** [comment]
Hello @yngccc,

I tried to look into it but couldn't reproduce the issue here.
Give your description, this alone should exhibit the issue but it doesn't:
```
ImGui::Begin("A");
ImGui::End();
ImGui::Begin("B");
ImGui::End();
```

Could you:
- provide you exact version (git hash)
- show the exact call stack
- create repro code for it or provide more information?

Thank you!

(Linking to #3163, #3129, #3232 for my own tracking)

**Comment by yngccc** [comment]
I just take the example code and comment out line 123-158 on _main.cpp_ and replace with

```
ImGui::Begin("A");
ImGui::End();
ImGui::Begin("B");
ImGui::End();
```

commit 2665288b30bfc7ac130699b791598d7753544db4

```
example_win32_directx11.exe!ImDrawList::_OnChangedClipRect() Line 488	C++
example_win32_directx11.exe!ImDrawList::PushClipRect(ImVec2 cr_min={...}, ImVec2 cr_max={...}, bool intersect_with_current_clip_rect=true) Line 556	C++
example_win32_directx11.exe!ImGui::PushClipRect(const ImVec2 & clip_rect_min={...}, const ImVec2 & clip_rect_max={...}, bool intersect_with_current_clip_rect=true) Line 4395	C++
example_win32_directx11.exe!ImGui::UpdateWindowManualResize(ImGuiWindow * window=0x000001906988d350, const ImVec2 & size_auto_fit={...}, int * border_held=0x00000071721ae7f4, int resize_grip_count=2, unsigned int * resize_grip_col=0x00000071721ae818) Line 5491	C++
example_win32_directx11.exe!ImGui::Begin(const char * name=0x00007ff7a0a07894, bool * p_open=0x0000000000000000, int flags=0) Line 6339	C++
example_win32_directx11.exe!main(int __formal=1, char * * __formal=0x00000190667e8f60) Line 126	C++
```

**Comment by yngccc** [comment]
Although this bug only happen if both windows are completely empty like in the code above.

**Comment by ocornut** [comment]
> I just take the example code and comment out line 123-158 on main.cpp and replace with

This is exactly what I did and I cannot repro yet, but still investigating how it could happen.

**Comment by ocornut** [comment]
OK I can repro the issue now, I got confused because your GIF showed an assert but I don't get one (although it does access uninitialized memory). Fixing now.

**Comment by ocornut** [comment]
Should be fixed by 1a1dcea1a02c4e1542b6e2619111fe2d20e22383 (master) and 76e40fe5d1059733ddcdd88236ee5c5743ed64b7 (docking)
Thanks for the report!
